sorting/selection_sort.cpp

```diff
@@ -1,33 +1,126 @@
-// Selection Sort
+/******************************************************************************
+ * @file
+ * @brief Implementation of the [Selection
+ *sort](https://en.wikipedia.org/wiki/Selection_sort) implementation using
+ *swapping
+ * @details
+ * The selection sort algorithm divides the input array into two parts: a sorted
+ * subarray of items which is built up from left to right at the front (left) of
+ * the array, and a subarray of the remaining unsorted items that occupy the
+ *rest of the array. Initially, the sorted subarray is empty, and the unsorted
+ *subarray is the entire input array. The algorithm proceeds by finding the
+ *smallest (or largest, depending on the sorting order) element in the unsorted
+ *subarray, exchanging (swapping) it with the leftmost unsorted element (putting
+ *it in sorted order), and moving the subarray boundaries one element to the
+ *right.
+ *
+ * ### Implementation
+ *
+ * SelectionSort
+ * The algorithm divides the input array into two parts: the subarray of items
+ * already sorted, which is built up from left to right. Initially, the sorted
+ * subarray is empty and the unsorted subarray is the entire input array. The
+ * algorithm proceeds by finding the smallest element in the unsorted subarray,
+ * exchanging (swapping) it with the leftmost unsorted element (putting it in
+ * sorted order), and moving the subarray boundaries one element to the right.
+ *
+ * @author [Lajat Manekar](https://github.com/Lazeeez)
+ *******************************************************************************/
+#include <algorithm>  ///for std::is_sorted
+#include <array>      ///for std::array
+#include <cassert>    ///for assert
+#include <iostream>   ///for io operations
 
-#include <iostream>
-using namespace std;
+/******************************************************************************
+ * @namespace sorting
+ * @brief Sorting algorithms
+ *******************************************************************************/
+namespace sorting {
 
-int main() {
-    int Array[6];
-    cout << "\nEnter any 6 Numbers for Unsorted Array : ";
-
-    // Input
-    for (int i = 0; i < 6; i++) {
-        cin >> Array[i];
-    }
+/******************************************************************************
+ * @brief The main function which implements Selection sort
+ * @param nums array to be sorted:-> array datatype, array length,
+ * @param len length of array to be sorted,
+ * @returns void
+ *******************************************************************************/
 
-    // Selection Sorting
-    for (int i = 0; i < 6; i++) {
-        int min = i;
-        for (int j = i + 1; j < 6; j++) {
-            if (Array[j] < Array[min]) {
-                min = j;  // Finding the smallest number in Array
+void selectionSort(
+    std::array<int, 8> &arr,
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). 🙂